avocado/utils/memory.py

```diff
@@ -27,7 +27,95 @@
 from . import genio
 
 
-# Returns total memory in kb
+class MemoryError(Exception):
+    """
+    called when memory operations fails
+    """
+    pass
+
+
+def _check_memory_state(block):
+    """
+    Check the given memory block is online or offline
+
+    :param block: memory block id.
+    :type string: like 198
+    :return: 'True' if online or 'False' if offline
+    :rtype: bool
+    """
+    while True:
+        try:
+            state = open(
+                '/sys/devices/system/memory/memory%s/state' % block).read()
+            if state not in ('going-online', 'going-offline'):
+                break
+        except Exception:
+            pass
```

Comment:
Why to pass? If you are trying to avoiding a known case, be explicit about the exception you're handling and let the other exceptions to break the execution so users will know what's happening.